MPCHomeControl/src/tools/mod.rs

```diff
@@ -0,0 +1,34 @@
+extern crate nalgebra as na;
+
+use chrono::prelude::*;
+use na::{Dot, Norm, Vector3};
+use spa;
+
+pub fn calc_solar_vector(azimuth: f64, zenith_angle: f64) -> Vector3<f64> {
+    let azimuth = azimuth.to_radians();
+    let elevation_deg = 90_f64 - zenith_angle;
+    let elevation = elevation_deg.to_radians();
+    let x = elevation.cos() * azimuth.cos();
+    let y = elevation.cos() * azimuth.sin();
+    let z = elevation.sin();
+    Vector3::new(x, y, z)
+}
+
+pub fn get_iluminated_area(
```

Comment:
Ten nazev myslim neni presny -- neni to illuminated area ale spis nejake effective illuminated area... nebo ne?